tools/common/error.cs

```diff
@@ -0,0 +1,89 @@
+// Copyright 2011-2013, Xamarin Inc. All rights reserved,
+// Copyright 2020 Microsoft Corp
+
+using System;
+using System.Text;
+
+namespace Xamarin.Bundler {
+
+#if MTOUCH || MONOTOUCH
+	public class MonoTouchException : Exception {
+		public const string Prefix = "MT";
+
+		public MonoTouchException (int code, string message, params object[] args) : 
+			this (code, false, message, args)
+		{
+		}
+
+		public MonoTouchException (int code, bool error, string message, params object[] args) : 
+			this (code, error, null, message, args)
+		{
+		}
+
+		public MonoTouchException (int code, bool error, Exception innerException, string message, params object[] args) : 
+			base (Format (message, args), innerException)
+		{
+			Code = code;
+			Error = error || ErrorHelper.GetWarningLevel (code) == ErrorHelper.WarningLevel.Error;
+		}
+#else
+	public class MonoMacException : Exception {
+		public const string Prefix = "MM";
+
+		public MonoMacException (int code, string message, params object[] args) :
```

Comment:
One way to make this safer is to add an overload without the `params` (for every overload with `params`, so three more constructors). This way passing a verbatim string works, because it won't end up calling string.Format.

Reply:
good idea 👍